### ISSUE TITLE
Bug: [return-await] allows function without return

### ISSUE THREAD

**Issue by devuxer** [typescript-eslint/typescript-eslint#8651]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.2&fileType=.tsx&code=PTAEDkHtQUwJzpOpIGNUFc4GdQDMlQAXACwEtsAaWADwAcZUiYATUAQwDtYEkAuALAAoGPSREO2AJ6dU%2BDLKJlI3InCkAVdQEEA5uzKcAFAEpQAb2GgOAdwMSACogC2FGADo4MbJAA2ANxhTAG5QEFBsEkgMXzYAIxgObnhEZC42PDIaYmgvIixudjsyRxc3T28-QKMAIn92XwwYGpNhAF8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaaRAF1ml0IEMB3agS1PSnwDM3IAacbGRAI6wGJGvSYtI7Tj2xQAAqQCeABxQBjaAxWlCKeA1ykA9CSEjEYxswyRE0aPmjdeORao1ade5AaOmyCio6axYAbVc%2Be0dnWTkoanh6JVRIgF1eAF8QTKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// No error occurs for this, expected an error:
export async function tryTryAgain() {
    await Promise.resolve(); // should be an error and fix to return await Promise.resolve("value")
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
     "no-return-await": "off",
     "require-await": "off",
     "@typescript-eslint/require-await": "error",
     "@typescript-eslint/return-await": ["error", "always"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Expected an error

### Actual Result

No error, no autofix to `return await`

### Additional Info

I do get an error and fix if I put only `return Promise.resolve("value")`.

**Comment by Josh-Cena** [comment]
This rule allows/disallows returned promises to be awaited. It doesn't say anything about non-returned promises. Why would you want this code to be flagged, anyway?

**Comment by bradzacher** [comment]
Josh is correct - this rule doesn't *enforce* a promise is returned only that a returned promise is or isn't awaited. Closing as working as intended.

Feel free to continue discussing though and we can reopen if required.

**Comment by devuxer** [comment]
@bradzacher,

Fair enough that I misinterpreted what this rule was supposed to do, but I keep reading and re-reading the [documentation](https://typescript-eslint.io/rules/return-await/), and I'm not convinced it actually says what this rule does.

What is says it does:
> Enforce consistent returning of awaited values.

What it actually does:
> Enforce consistent awaiting of returned values.

Wordier but more complete version:
> Enforce whether a returned promise is always awaited, never awaited, or awaited when inside a try/catch.

The reason I thought I wanted it to actually return the awaited value is to enable better stack traces, but I acknowledge I don't quite get the exact circumstances under which it's necessary to return the awaited expression to attain that benefit. It would be nice if the documentation made the stack trace behavior for the different options of this rule crystal clear, as it's difficult to really make an intelligent decision about this rule's options without this knowledge. 

For what it's worth, my codebase currently uses `return await` for getters/queries and just `await` for setters/mutations.

**Comment by bradzacher** [comment]
The wording in the docs is a bit cumbersome - happy to accept a PR if you think you can improve the docs.

**Comment by devuxer** [comment]
@bradzacher, my previous comment is a pretty good preview of how I think the docs could be improved. I'm happy to work on a pull request, but I would appreciate some feedback on the two different wording options I offered as well as some help with my question about stack trace behavior
